datastream-server/src/main/java/com/linkedin/datastream/server/Coordinator.java

```diff
@@ -663,7 +663,11 @@ private void assignSerdes(DatastreamTaskImpl datastreamTask) {
 
   protected synchronized void handleEvent(CoordinatorEvent event) {
     _log.info("START: Handle event " + event.getType() + ", Instance: " + _adapter.getInstanceName());
-
+    boolean isLeader = _adapter.isLeader();
+    if (skipLeaderEvent(event.getType(), isLeader)) {
```

Comment:
nit: I think this'd be more intuitive/straightforward if expressed as:
```java
if (isLeaderEvent(event) && !isLeader) {
  _log.info(...);
  return;
}
```

Reply:
done